src/goose/synopsis/moderator.py

```diff
@@ -91,9 +94,12 @@ def get_synopsis(self, exchange: Exchange, summarize: bool = False, plan: bool =
     def summarize(self, exchange: Exchange) -> str:
         message = Message.load("summarize.md", synopsis=self, messages=self.originals, exchange=exchange, system=system)
         model = os.environ.get("SUMMARIZER", exchange.model)
-        new_exchange = exchange.replace(moderator=ContextTruncate(), tools=(), system="", messages=[], model=model)
-        new_exchange.add(message)
-        return new_exchange.generate().content[0].text
+        if self.rewrites % 20 == 0:
+            new_exchange = exchange.replace(moderator=ContextTruncate(), tools=(), system="", messages=[], model=model)
+            new_exchange.add(message)
+            return new_exchange.generate().content[0].text
+        else:
+            return message
```

Comment:
i think this ends up violating our return type `str` this looks like type `Message`. do we want to return a both or just the content (e.g. `self.current_summary`)? i'm thinking the latter?

Reply:
yeah - I think I will close this in favour of: https://github.com/block/goose/pull/190 - which does this correctly (as well as update status) and doesn't have this bug